File: backend/src/main/java/com/codeloom/backend/dao/testcase/TestCaseRepositoryCustom.java
package com.codeloom.backend.dao.testcase;

import com.codeloom.backend.model.TestCase;
import java.util.Collection;

public interface TestCaseRepositoryCustom {
  Collection<TestCase> findAllByProblemId(long id, Boolean isPublic);

  int countAllByProblemId(long id, Boolean isPublic);

  default int countAllByProblemId(long id) {
    return countAllByProblemId(id, null);
  }
}

File: backend/src/main/kotlin/com/codeloom/backend/dao/testcase/TestCaseRepositoryCustom.kt
package com.codeloom.backend.dao.testcase

import com.codeloom.backend.model.TestCase

interface TestCaseRepositoryCustom {
    fun findAllByProblemId(
        problemId: Long,
        isPublic: Boolean? = null,
    ): Collection<TestCase>

    fun countAllByProblemId(
        problemId: Long,
        isPublic: Boolean? = null,
    ): Int
}
